### PR TITLE
DOC: Improved docs for `jnp.nansum` and `nanprod`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461.

### CODE REVIEW

File: jax/_src/numpy/reductions.py [hunk 1]
```diff
@@ -1200,25 +1200,167 @@ def nanmax(a: ArrayLike, axis: Axis = None, out: None = None,
...
 def nansum(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None, out: None = None,
            keepdims: bool = False, initial: ArrayLike | None = None,
            where: ArrayLike | None = None) -> Array:
+  r"""Return the sum of the array elements along a given axis, ignoring NaNs.
+
+  JAX implementation of :func:`numpy.nansum`.
+
+  Args:
+    a: Input array.
+    axis: int or array, default=None. Axis along which the sum is computed. If
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1214]:
```suggestion
    axis: int or sequence of ints, default=None. Axis along which the sum is computed. If
```

File: jax/_src/numpy/reductions.py [hunk 2]
```diff
@@ -1200,25 +1200,167 @@ def nanmax(a: ArrayLike, axis: Axis = None, out: None = None,
...
+      None, the sum is computed along the flattened array.
+    dtype: The type of the output array. Default=None.
+    keepdims: bool, default=False. If True, reduced axes are left in the result
+      with size 1.
+    initial: int or array, default=None. Initial value for the sum.
+    where: int or array of boolean dtype, default=None. The elements to be used
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1220]:
```suggestion
    where: array of boolean dtype, default=None. The elements to be used
```

File: jax/_src/numpy/reductions.py [hunk 3]
```diff
@@ -1200,25 +1200,167 @@ def nanmax(a: ArrayLike, axis: Axis = None, out: None = None,
...
+      given axis, ignoring NaNs.
+    - :func:`jax.numpy.nanmean`: Compute the mean of array elements along a given
+      axis, ignoring NaNs.
+
+  Examples:
+
+    By default, ``jnp.nansum`` computes the sum of elements along the flattened
+    array.
+
+    >>> nan= jnp.nan
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1243]:
```suggestion
    >>> nan = jnp.nan
```

File: jax/_src/numpy/reductions.py [hunk 4]
```diff
@@ -1200,25 +1200,167 @@ def nanmax(a: ArrayLike, axis: Axis = None, out: None = None,
...
+
+  Args:
+    a: Input array.
+    axis: int or array, default=None. Axis along which the product is computed.
+      If None, the product is computed along the flattened array.
+    dtype: The type of the output array. Default=None.
+    keepdims: bool, default=False. If True, reduced axes are left in the result
+      with size 1.
+    initial: int or array, default=None. Initial value for the product.
+    where: int or array of boolean dtype, default=None. The elements to be used
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1303]:
```suggestion
    where: array of boolean dtype, default=None. The elements to be used
```

File: jax/_src/numpy/reductions.py [hunk 5]
```diff
@@ -1200,25 +1200,167 @@ def nanmax(a: ArrayLike, axis: Axis = None, out: None = None,
...
 def nanprod(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None, out: None = None,
             keepdims: bool = False, initial: ArrayLike | None = None,
             where: ArrayLike | None = None) -> Array:
+  r"""Return the product of the array elements along a given axis, ignoring NaNs.
+
+  JAX implementation of :func:`numpy.nanprod`.
+
+  Args:
+    a: Input array.
+    axis: int or array, default=None. Axis along which the product is computed.
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1297]:
```suggestion
    axis: int or sequence of ints, default=None. Axis along which the product is computed.
```

File: jax/_src/numpy/reductions.py [hunk 6]
```diff
@@ -1200,25 +1200,167 @@ def nanmax(a: ArrayLike, axis: Axis = None, out: None = None,
...
+      axis, ignoring NaNs.
+    - :func:`jax.numpy.nanmean`: Compute the mean of array elements along a given
+      axis, ignoring NaNs.
+
+  Examples:
+
+    By default, ``jnp.nanprod`` computes the product of elements along the flattened
+    array.
+
+    >>> nan= jnp.nan
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1326]:
```suggestion
    >>> nan = jnp.nan
```

**Review by dfm — CHANGES_REQUESTED** [review]
Thanks! I left a few small comments, otherwise looks great!

File: jax/_src/numpy/reductions.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 5 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 6 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

**Comment by rajasekharporeddy** [comment]
Modified the docstrings and squashed the commits into one.

**Review by dfm — APPROVED** [review]

**Comment by dfm** [comment]
Since #22366 was merged, it looks like we can't merge until you rebase onto `main`. Can you do that now? Thanks!

**Comment by rajasekharporeddy** [comment]
Rebased onto main.

**Pull request merged**